admin-ui/src/component/reservations/requested/VisibleIfPermission.tsx

```diff
@@ -1,49 +1,30 @@
 import React from "react";
-import { useAuthState } from "../../../context/AuthStateContext";
-
-type Props = {
-  unitPk: number;
-  serviceSectorPks: number[];
-  permissionName: string;
-  children: React.ReactNode;
-  otherwise?: React.ReactNode;
-};
+import { type ReservationType } from "common/types/gql-types";
+import { usePermission } from "./hooks";
 
 const VisibleIfPermission = ({
-  unitPk,
-  serviceSectorPks,
+  reservation,
   permissionName,
   children,
   otherwise,
-}: Props): JSX.Element => {
-  const { hasPermission } = useAuthState().authState;
-  const permission = hasPermission(permissionName, unitPk, serviceSectorPks);
-
-  return (
-    <>
-      {permission && children}
-      {!permission && otherwise}
-    </>
-  );
-};
-
-const VisibleIfPermissionWrapper = (props: {
-  unitPk?: number;
-  serviceSectorPks: number[];
-  permissionName: string;
+}: {
+  reservation: ReservationType;
+  permissionName:
+    | "can_manage_reservations"
+    | "can_comment_reservations"
+    | "can_view_reservations";
```

Comment:
Should we consider a case where the user would need a combination of multiple permissions to access content?

Reply:
I think this component is a temporary solution, if there is direct requirement for multiple permission I'll add it then.
The proper solution would be to refactor the AuthContext and move the new permission check hook there, then maybe improve the VisibleIfPermission component if the hook isn't good enough.
The context uses reducer for data and an outdated Auth package so trying not to touch it too much :S